src/main/java/updater/JarUpdater.java

```diff
@@ -0,0 +1,255 @@
+                log("Wait for a while before trying again.");
+                Thread.sleep(WAIT_TIME);
+            } catch (InterruptedException e) {
+                log("Failed to wait for a while");
+            }
+        }
+
+        return false;
+    }
+
+    private boolean moveFileAsBackup(File file) {
+        log("Moving file as backup");
+        String backupFilename = getBackupFilename(file.getName());
+        File backup = new File(file.getParent(), backupFilename);
+        return moveFile(file.toPath(), backup.toPath());
+    }
+
+    private boolean restoreBackup(File file) {
+        log("Restoring backup");
+        String backupFilename = getBackupFilename(file.getName());
+        File backup = new File(file.getParent(), backupFilename);
+        return moveFile(backup.toPath(), file.toPath());
+    }
+
+    private void removeBackup(File file) {
+        log("Removing backup");
+        String backupFilename = getBackupFilename(file.getName());
+        File backup = new File(file.getParent(), backupFilename);
+        if (backup.exists() && !backup.delete()) {
+            log("Failed to remove backup");
+        }
+    }
+
+    private String getBackupFilename(String filename) {
+        return filename + BACKUP_FILENAME_SUFFIX;
+    }
+
+    private boolean createDirectories(File dir) {
+        if (!dir.exists() && !dir.mkdirs()) {
+            log("Failed to create directories for file " + dir.getName());
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Runs a command to execute a JAR file with the given arguments and options
+     * @param jarPath path (in string) to JAR file to be executed
+     * @param argsAndOptions raw arguments and options to run the JAR
+     * @return true if JAR can be executed, false otherwise
+     */
+    private static boolean executeJar(String jarPath, String argsAndOptions) {
+        String command = String.format("java -jar %1$s %2$s", jarPath, argsAndOptions).trim();
+        log("Starting JAR with command: " + command);
+
+        Process process;
+
+        try {
+            process = Runtime.getRuntime().exec(command);
+        } catch (IOException e) {
+            log("Failed to run command. Error: " + e.toString());
+            return false;
+        }
+
+        return process.isAlive();
+    }
+
+    /**
+     * Not using Apache Logger since it will require another initialization
+     */
+    private void initLogger() {
+        try {
+            PrintStream out = new PrintStream(new FileOutputStream(new File("update.log")),
+                    true, "UTF-8");
+            System.setOut(out);
+        } catch (FileNotFoundException e) {
+            log("File not found, will not create logger");
+        } catch (UnsupportedEncodingException e) {
+            log("Encoding not supported, will not create logger");
+        }
+    }
+
+    private static void log(String message) {
+        System.out.println(message);
```

Comment:
Because I am compiling this on its own, all the libraries and dependencies are not inside, so logger is not included, which is why I created the `log` method. I tried to use logger but I didn't manage to make the gradle task to include it. Should I try again?